### PR TITLE
feat(interview-prep): Porting Rosetta problems

### PR DESCRIPTION
**Pull request by NitronR**
#### Pre-Submission Checklist

- [ ] Your pull request targets the `dev` branch.
- [ ] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/challenge-tests`)
- [ ] All new and existing tests pass the command `npm test`.
- [ ] Use `npm run commit` to generate a conventional commit message.
    Learn more here: <https://conventionalcommits.org/#why-use-conventional-commits>
- [ ] The changes were done locally on your machine and NOT GitHub web interface.
    If they were done on the web interface you have ensured that you are creating conventional commit messages.

#### Checklist:

- [ ] Tested changes locally.
- [ ] Addressed currently open issue (replace XXXXX with an issue no in next line)

Closes #XXXXX

### CODE REVIEW

File: challenges/08-coding-interview-prep/rosetta-code.json [hunk 1]
```diff
@@ -15319,6 +15319,1381 @@
...
+        }
+      }
+    },
+    {
+      "title": "Rep-string",
+      "description": [
+        "Given a series of ones and zeroes in a string, define a repeated string or <i>rep-string</i> as a string which is created by repeating a substring of the <i>first</i> N characters of the string <i>truncated on the right to the length of the input string, and in which the substring appears repeated at least twice in the original</i>.",
+        "For example, the string <b>10011001100</b> is a rep-string as the leftmost four characters of <b>1001</b> are repeated three times and truncated on the right to give the original string.",
+        "Note that the requirement for having the repeat occur two or more times means that the repeating unit is <i>never</i> longer than half the length of the input string.",
+        "Write a function that takes a string and returns the string that is repeated."
```
Comment (scissorsneedfoodtoo) [inline challenges/08-coding-interview-prep/rosetta-code.json]: "Write a function that takes a string and returns the <code>rep-string</code>."

File: challenges/08-coding-interview-prep/rosetta-code.json [hunk 2]
```diff
@@ -15319,6 +15319,1381 @@
...
+          "contents": ["function repString (str) {", "  // Good luck!", "}"],
+          "head": [],
+          "tail": []
+        }
+      }
+    },
+    {
+      "title": "Resistor mesh",
+      "description": [
+        "<a href=\"http://rosettacode.org/wiki/image:resistor-mesh.svg\">300px||right</a>",
```
Comment (scissorsneedfoodtoo) [inline challenges/08-coding-interview-prep/rosetta-code.json]: `"Given   10&times;10   grid nodes  (as shown in the image)  interconnected by   1Ω   resistors as shown, write a function to find the resistance between point  <b>A</b>  and  <b>B</b>.`

Also, is there any way to get the image this challenge is referring to?

File: challenges/08-coding-interview-prep/rosetta-code.json [hunk 3]
```diff
@@ -15319,6 +15319,1381 @@
...
+          "tail": []
+        }
+      }
+    },
+    {
+      "title": "Roots of a function",
+      "description": [
+        "Create a function that finds and returns the roots of a given function as an array.",
+        "In this task, you need to use brute force and put values in the given function to check if it is a root.",
+        "The function should take an array and 3 numbers as parameters. The array represents the function. The $ i^{th} $ element is the coefficient of $ x^i $. The numbers are lower bound, upper boand and step values. You need to start from the lower bound, increament by step value until the upper bound is reached. The values obtained should be tested if they are the roots of the given function."
```
Comment (scissorsneedfoodtoo) [inline challenges/08-coding-interview-prep/rosetta-code.json]: `"The function should take an array and 3 numbers as parameters. The array represents the function. The $ i^{th} $ element is the coefficient of $ x^i $. The numbers are the lower bound, upper bound, and step values, respectively. You need to start from the lower bound and increment by the step value until the upper bound is reached. The values obtained should be tested to see if they are the roots of the given function."`

File: challenges/08-coding-interview-prep/rosetta-code.json [hunk 4]
```diff
@@ -15319,6 +15319,1381 @@
...
+      "solutions": [
+        "function roots (poly, lowerBound, upperBound, step) {\n  function sign(x) {\n    return (x < 0.0) ? -1 : (x > 0.0) ? 1 : 0;\n  }\n\n  function f(x) {\n    var val=0;\n    poly.forEach(function (e,i) {\n      val+=Math.pow(x,i)*e;\n    })\n    return val;\n  }\n\n  var x = lowerBound,\n    ox = x,\n    y = f(x),\n    oy = y,\n    s = sign(y),\n    os = s;\n  var roots=[]\n  for (; x <= upperBound; x += step) {\n    s = sign(y = f(x));\n    if (s == 0) {\n    } else if (s != os) {\n      var dx = x - ox;\n      var dy = y - oy;\n      var cx = x - dx * (y / dy);\n      roots.push(Math.round(cx*1000)/1000)\n    }\n    ox = x;\n    oy = y;\n    os = s;\n  }\n\n  return roots\n}\n"
+      ],
+      "tests": [
+        {
+          "text": "'<code>roots</code> should be a function.'",
+          "testString":
+            "assert(typeof roots=='function','<code>roots</code> should be a function.');"
+        },
+
```
Comment (scissorsneedfoodtoo) [inline challenges/08-coding-interview-prep/rosetta-code.json]: `"Given a string containing uppercase characters (A-Z), compress repeated 'runs' of the same character and store the length of that run by completing the <code>encode</code> function, and complete the <code>decode</code> function to reverse the compression.",`

Comment (scissorsneedfoodtoo) [inline challenges/08-coding-interview-prep/rosetta-code.json]: This and the following line can be edited to be more specific to this challenge. Maybe something like:

`"Example:",
"encode('WWWWWWWWWWWWBWWWWWWWWWWWWBBBWWWWWWWWWWWWWWWWWWWWWWWWBWWWWWWWWWWWWWW') should return <code>[[12,\"W\"],[1,\"B\"],[12,\"W\"],[3,\"B\"],[24,\"W\"],[1,\"B\"],[14,\"W\"]]</code>."`

We would also need a simple example for the decode function.

**Comment by NitronR** [comment]
@scissorsneedfoodtoo Sorry for the delay. I have updated the descriptions. I will look into adding the image for the challenge mentioned by you.

**Comment by NitronR** [comment]
@scissorsneedfoodtoo I have added the image. Please check it and tell me if anything else is needed to be done.

**Comment by scissorsneedfoodtoo** [comment]
@NitronR, everything LGTM! Thank you for taking the time to make all these changes. Merging now :+1: :+1:

**Pull request merged**